internal/utils/codecrafters_client.go

```diff
@@ -88,13 +88,20 @@ func (c CodecraftersClient) headers() map[string]string {
 	}
 }
 
-func (c CodecraftersClient) CreateSubmission(repositoryId string, commitSha string) (CreateSubmissionResponse, error) {
+func (c CodecraftersClient) CreateSubmission(repositoryId string, commitSha string, previous bool) (CreateSubmissionResponse, error) {
+	requestBody := map[string]interface{}{
+		"repository_id":            repositoryId,
+		"commit_sha":               commitSha,
+		"should_auto_advance":      false,
+		"stage_selection_strategy": "current_and_previous_descending",
+	}
+
+	if previous {
```

Comment:
This decision should be one layer above, in the caller. A "client" for core shouldn't be aware of anything other than what core supports. We should accept `stageSelectionStrategy` here, and have the caller decide what value to pass in.